src/components/timezone/DateAndTime.jsx

```diff
@@ -0,0 +1,458 @@
+            }
+        };
+
+        updateTimeDateData();
+    }, [timezone]);
+
+    const handleAutoDateTime = useCallback(() => {
+        const newValue = !autoDateTime;
+        setAutoDateTime(newValue);
+        setNTPEnabled({ enabled: newValue });
+    }, [autoDateTime]);
+
+    // Section validity update
+    useEffect(() => {
+        setSectionValid(autoDateTime || datetime.isValid);
+    }, [autoDateTime, datetime, setSectionValid]);
+
+    const handleDateChange = (_event, _, newDate) => {
+        if (!isValidDate(newDate)) return;
+
+        const localDate = DateTime.fromJSDate(newDate);
+        const datePart = localDate.setZone(timezone, { keepLocalTime: true });
+
+        const combined = datetime.set({
+            day: datePart.day,
+            month: datePart.month,
+            year: datePart.year,
+        });
+
+        setDatetime(combined);
+
+        if (!autoDateTime && combined.isValid) {
+            setSystemDateTime({ dateTimeSpec: combined.toISO() });
+        }
+    };
+
+    // TimePicker change handler
+    const handleTimeChange = (event, _, hour, minute, seconds, isValid) => {
+        if (!isValid) return;
+        if (typeof hour !== "number" || typeof minute !== "number") return;
+        const updated = datetime.set({
+            hour,
+            minute,
+            second: seconds || 0,
+        });
+        setDatetime(updated);
+        if (!autoDateTime && updated.isValid) {
+            setSystemDateTime({ dateTimeSpec: updated.toISO() });
+        }
+    };
+
+    const handleNtpServerSelect = (event, value) => {
+        setNtpServer(value);
+        setNtpSelectOpen(false);
+        setTimeSources({ sources: [{ enabled: true, name: value, type: "NTP" }] });
+    };
+
+    const ntpToggle = toggleRef => (
+        <MenuToggle
+          id={`${SCREEN_ID}-ntp-toggle`}
+          ref={toggleRef}
+          onClick={() => setNtpSelectOpen(!ntpSelectOpen)}
+          isExpanded={ntpSelectOpen}
+          isDisabled={!autoDateTime}
+        >
+            {ntpServer}
+        </MenuToggle>
+    );
+
+    // Always derive these on render
+    const datePickerValue = formatDateInput(datetime);
+    const timePickerValue = getLocalTimeForPicker(datetime, timeFormat === "24");
+
+    return (
+        <>
+            <Title headingLevel="h2">{_("Date and time")}</Title>
+            <FormGroup>
+                <Stack hasGutter>
+                    <StackItem>
+                        <Flex alignItems={{ default: "alignItemsCenter" }}>
+                            <FlexItem>
+                                <Checkbox
+                                  id={`${SCREEN_ID}-auto-date-time`}
+                                  isChecked={autoDateTime}
+                                  onChange={handleAutoDateTime}
+                                  label={_("Automatically set date and time, using time server:")}
+                                  aria-label={_("Automatically set date and time")}
```

Comment:
If you're already using a visible label like here, adding aria-label is redundant and can confuse screen readers. 
Please remove.